network/ovs_networkclient_linux.go

```diff
@@ -55,56 +50,34 @@ func updateOVSConfig(option string) error {
 	return nil
 }
 
-func NewOVSClient(bridgeName, hostInterfaceName, snatBridgeIP string, skipAddressesFromBlock []string, enableSnatOnHost bool) *OVSNetworkClient {
+func NewOVSClient(bridgeName, hostInterfaceName string) *OVSNetworkClient {
 	ovsClient := &OVSNetworkClient{
-		bridgeName:             bridgeName,
-		hostInterfaceName:      hostInterfaceName,
-		snatBridgeIP:           snatBridgeIP,
-		skipAddressesFromBlock: skipAddressesFromBlock,
-		enableSnatOnHost:       enableSnatOnHost,
+		bridgeName:        bridgeName,
+		hostInterfaceName: hostInterfaceName,
 	}
 
 	return ovsClient
 }
 
 func (client *OVSNetworkClient) CreateBridge() error {
-	if err := ovsctl.CreateOVSBridge(client.bridgeName); err != nil {
-		return err
-	}
+	var err error
 
-	if err := updateOVSConfig(ovsOpt); err != nil {
+	if err = ovsctl.CreateOVSBridge(client.bridgeName); err != nil {
 		return err
 	}
 
-	if client.enableSnatOnHost {
-		if err := ovssnat.CreateSnatBridge(client.snatBridgeIP, client.bridgeName); err != nil {
-			log.Printf("[net] Creating snat bridge failed with erro %v", err)
-			return err
+	defer func() {
+		if err != nil {
+			client.DeleteBridge()
 		}
+	}()
 
-		if err := ovssnat.AddMasqueradeRule(client.snatBridgeIP); err != nil {
-			return err
-		}
-
-		return ovssnat.AddVlanDropRule()
-	}
-
-	return nil
+	return updateOVSConfig(ovsOpt)
```

Comment:
does the defer func care about the error in the return func?

Reply:
yes